### PR TITLE
Incorporated feedback as discussed with Jeroen

### PR DESCRIPTION
**Pull request by DrRetsiemsuah**
## Structure

**Section 2**
- Merged "Remove Human Interaction" (2.4) with 2.8 to "2.4 Automate Secrets Management"
- Merge "Use Least Privilege (2.4)" with 2.3 to "2.3 Access Controls"

**Section 4**
- Added TCBs for additional secret storage services from Cloud providers
    - AWS Nitro Enclaves
    - AWS CloudHSM
    - Google Cloud Confidential Computing
    - Azure Confidential Computing
    - Azure Dedicated HSM 

## Content
**Section 2.6.2**
- Added exclusion for user credentials according to e.g. NIST guidelines nowadays suggest that passwords should only be rotated, if there is evidence for compromise ([NIST Special Publication 800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html#:~:text=Do%20not%20require%20that%20memorized%20secrets%20be%20changed%20arbitrarily%20(e.g.%2C%20periodically)%20unless%20there%20is%20a%20user%20request%20or%20evidence%20of%20authenticator%20compromise.%20(See%20Section%205.1.1%20for%20additional%20information)))

**Section 2.11**
- Added metadata “By whom it was created/consumed/archived/rotated/deleted” (referring to Auditing requirements)

**Section 3.2**
- Added GCP Secret Manager ([Secret Manager | Google Cloud](https://cloud.google.com/secret-manager))

### CODE REVIEW

**Comment by commjoen** [comment]
Maybe we can do this one after https://github.com/OWASP/CheatSheetSeries/pull/841 so we can do this in steps?

**Comment by mackowski** [comment]
This now have conflicts, please fix the conflicts.

File: cheatsheets_draft/Secrets_Management_CheatSheet.md [hunk 1]
```diff
@@ -248,6 +243,17 @@ It is also possible to use the [Systems Manager Parameter store](https://docs.aw
 - it doesn't support cross-region replication
 - there are fewer [security hub controls](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html) available
 
+##### 4.1.1.1 AWS Nitro Enclaves
+
+- TODO: https://aws.amazon.com/ec2/nitro/nitro-enclaves/
+
+##### 4.1.1.2 AWS CloudHSM
+
+- TODO: https://aws.amazon.com/cloudhsm/
```
Comment (commjoen) [inline cheatsheets_draft/Secrets_Management_CheatSheet.md:252]: Maybe for now first detail/fill in the todos before this gets merged?

File: cheatsheets_draft/Secrets_Management_CheatSheet.md [hunk 2]
```diff
@@ -248,6 +243,17 @@ It is also possible to use the [Systems Manager Parameter store](https://docs.aw
 - it doesn't support cross-region replication
 - there are fewer [security hub controls](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html) available
 
+##### 4.1.1.1 AWS Nitro Enclaves
+
+- TODO: https://aws.amazon.com/ec2/nitro/nitro-enclaves/
```
Comment (commjoen) [inline cheatsheets_draft/Secrets_Management_CheatSheet.md:248]: Maybe for now first detail/fill in the todos before this gets merged?

File: cheatsheets_draft/Secrets_Management_CheatSheet.md [hunk 3]
```diff
@@ -256,6 +262,10 @@ Permissions are granted at the secret level.
 
 Check out the [Secret Manager best practices](https://cloud.google.com/secret-manager/docs/best-practices) for more information.
 
+##### 4.1.2.1 Google Cloud Confidential Computing
+
+TODO: https://cloud.google.com/confidential-computing
```
Comment (commjoen) [inline cheatsheets_draft/Secrets_Management_CheatSheet.md:267]: Maybe for now first detail/fill in the todos before this gets merged?

File: cheatsheets_draft/Secrets_Management_CheatSheet.md [hunk 4]
```diff
@@ -264,6 +274,14 @@ Contrary to other clouds, permisssions are granted at the _**Key Vault**_ level.
 
 Check out the [Key Vault best practices](https://docs.microsoft.com/en-us/azure/key-vault/general/best-practices) for more information.
 
+##### 4.1.3.1 Azure Confidential Computing
+
+TODO: https://azure.microsoft.com/en-us/solutions/confidential-compute/#overview
+
```
Comment (commjoen) [inline cheatsheets_draft/Secrets_Management_CheatSheet.md:280]: Maybe for now first detail/fill in the todos before this gets merged?

File: cheatsheets_draft/Secrets_Management_CheatSheet.md [hunk 5]
```diff
@@ -264,6 +274,14 @@ Contrary to other clouds, permisssions are granted at the _**Key Vault**_ level.
...
+##### 4.1.3.2 Azure Dedicated HSM
+
+TODO: https://azure.microsoft.com/en-us/services/azure-dedicated-hsm/
```
Comment (commjoen) [inline cheatsheets_draft/Secrets_Management_CheatSheet.md:283]: Maybe for now first detail/fill in the todos before this gets merged?

**Review by commjoen — CHANGES_REQUESTED** [review]
Can you please first resolve the conflicts and make sure there is the right content at the todos?

**Comment by DrRetsiemsuah** [comment]
@commjoen I'll work on resolving the todos later this week and update the PR

**Pull request closed without merging**